R/msf_dict_mortality.R

```diff
@@ -0,0 +1,76 @@
+# function to load MSF data dictionary for mortality surveys
+
+#' @importFrom rio import
+#' @importFrom epitrix clean_labels
+#' @importFrom tibble as_tibble
+#' @importFrom tidyr fill spread
+#' @importFrom dplyr mutate group_by row_number ungroup select recode
+#' @importFrom tidyselect everything
+#' @export
+#' @rdname msf_dict
+msf_dict_mortality <- function(name = "MSF-mortality_survey-dict.xlsx",
+                               tibble = TRUE) {
+
+  # get excel file path (need to specify the file name)
+  path <- system.file("extdata", name, package = "sitrep")
+
+  # read in data set - pasting the disease name for sheet
+  dat_dict <- rio::import(path, which = "Data Dictionary")
+
+  # clean col names
+  colnames(dat_dict) <- epitrix::clean_labels(colnames(dat_dict))
+
+  # fill NA values with previous non-NA value, replace "." in codes and names
+  dat_dict <- tidyr::fill(dat_dict,
+                          tidyselect::everything(), .direction = "down")
+
+  # minor tidying, e.g.: create "CodeX" assignments
+  dat_dict <- dplyr::mutate(dat_dict,
+    choice_code = replace(choice_code, choice_code == ".", NA),
```

Comment:
tidyverse operations like this are best replaced by base R equivalents in packages.

For instance, this should be

```r
dat_dict$choice_code[dat_dict$choice_code == "."] <- NA
dat_dict$choice_name[dat_dict$choice_name == "."] <- NA
```